macros/env_generate_schema_name.sql

```diff
@@ -0,0 +1,31 @@
+/* xdb: nocoverage */
+{% macro env_generate_schema_name(custom_schema_name, branch_name, default_schema) -%}
+{#/* Used in conjunction with generate_schema_name, this macro returns a schema name
+        based on the cusrrent working environment
+       ARGS:
+         - custom_schema_name (string) The configured value of schema in the specified node, or none if a value is not supplied
+         - branch_name (string) The current branch name
+         - default_schema (string) The default schema name e.g. target.schema
+       RETURNS: A schema name.
+       SUPPORTS:
+            - Postgres
+            - Snowflake
+    */#}  
+  {%- if branch_name == '' -%}
+    {%- if custom_schema_name is none -%}
+      {{ default_schema }}
+    {%- else -%}
+      {{ default_schema }}_{{ custom_schema_name | trim }}
+    {%- endif -%}
+  {%- else -%}
+    {%- if custom_schema_name is none -%}
+      {{ xdb._dev_schema(branch_name) }}_{{ default_schema }}
+    {%- else -%}
+      {{ xdb._dev_schema(branch_name) }}_{{ default_schema }}_{{ custom_schema_name | trim }}
+    {%- endif -%}
+  {%- endif -%}
+{%- endmacro %}
+
+{%- macro _dev_schema(branch_name) -%}
```

Comment:
Actually, one thought on this even though the PR looks good. It might be worth changing this macro name to be more descriptive. Something like 'normalize*', or 'replace*' might more sense than `_dev_schema` since it's just removing "bad" common characters (and lowercasing).

Reply:
update the name, thank you!